src/bin/imageprimaries.rs

```diff
@@ -34,28 +35,27 @@ use std::error::Error;
 use units::{Length, mm, mm_, todo::Lengthf32};
 use petalo::fov::FOV;
 use petalo::image::Image;
-use petalo::io::hdf5::{read_table, Primary};
+use petalo::io::mcreaders::{read_primaries, MCPrimary};
```

Comment:
I had thought of `io:hdf5` as the base where specific low level functions lived and that the readers for our types would go in an adjacent file. Would the name change `io::mcreaders` --> `io::h5datatypes` or similar work? Or do you want everything dumped into the original file?

As to MC vs. general, there are some which are obviously only MC (vertices etc) and others that it really depends on how things progress. For real data I'd expect to have to process a bit differently (no smearing etc) but whether that affects the structure on disc is another thing.

Reply:
> Would the name change `io::mcreaders` --> `io::h5datatypes` or similar work?

I don't like this because HDF5 is an implementation detail of how data are serialized/persisted (and other formats will be used, sooner or later). The reconstruction algorithm should care not one hoot about how the data were stored, while 'datatype' sounds like something that we want to reason about in our reconstruction code.

The whole of `io` in general is about getting data in/out of persistent storage. The reconstruction code should be decoupled and unaware of this as much as possible.

I don't necessarily want everything dumped back in the original file. Remember that Rust allows arbitrary-depth nesting of modules. I think there should be `io::hdf5::` and then organize things sensibly in submodules within that.

I would avoid having 'read' in the module name: we happen to be using it only for reading in these specific cases, but deriving `H5Type` provides writing capabilities too (we do write HDF5 in e.g. `makelor`). That's covered by the fact that it's a submodule of `io`.

We might already split things into `io::hdf5::mc` and `io::hdf5::xxx` where `xxx` would need to be some not-too-stupid name that means non-mc. It might not be worth splitting it until we know more about the details.

How about this:

+ Remove the `MC` prefix from all the names in `mcreaders.rs`
+ Move all those which are mc-specific to `io::hdf5::mc`, move the others to `io::hdf5` (or `io::hdf5::xxx`).

?